api/server_test.go

```diff
@@ -73,10 +76,10 @@ func TestLogger(t *testing.T) {
 }
 
 func TestWithEngine(t *testing.T) {
-	engine, err := core.NewEngine(nil, lib.Options{})
-	if !assert.NoError(t, err) {
-		return
-	}
+	executor, err := local.New(&lib.MiniRunner{}, logrus.StandardLogger())
+	require.NoError(t, err)
+	engine, err := core.NewEngine(executor, lib.Options{}, logrus.StandardLogger())
+	require.NoError(t, err)
```

Comment:
Maybe have this 4 lines of code as a separate function as this is the nth (might be kth as n is quite big but you get the point) time I see it

Reply:
This has only 4 repeats, with 1 in a different package than the rest, and at most 2 repeats in a single file. So IMO not worth over-complicating at this point....